tests/hla/process_randomlist.py

```diff
@@ -0,0 +1,204 @@
+import sys
+import traceback
+import os
+import datetime
+import pytest
+import numpy as np
+from astropy.table import Table, vstack
+from astropy.io import ascii
+
+from .base_test import BaseHLATest
+from drizzlepac import alignimages
+import drizzlepac.hlautils.catalog_utils as catutils
+
+class TestAlignMosaic(BaseHLATest):
+    """ Process a large sample of ACS and WFC3 datasets to determine if they can be
+        aligned to an astrometric standard.
+    """
+
+    def test_align_randomFields(self):
+        """ Wrapper to set up the test for aligning a large number of randomly
+            selected fields (aka datasets) from a input ascii file (CSV).
+
+            The wrapper provides the parameter settings for the underlying test,
+            as well as implements the criterion for the overall success or failure
+            of the test.
+        """
+
+        inputListFile = 'ACSWFC3List.csv'
+
+        # Desired number of random entries for testing
+        inputNumEntries = 50
+
+        # Seed for random number generator
+        inputSeedValue = 1
+
+        # Obtain the full path to the file containing the dataset field names
+        self.input_repo = 'hst-hla-pipeline'
+        self.tree = 'dev'
+        self.input_loc = 'master_lists'
+        input_file_path = self.get_data(inputListFile)
+
+        # Randomly select a subset of field names (each field represented by a row) from
+        # the master CSV file and return as an Astropy table
+        randomCandidateTable = catutils.randomSelectFromCSV(input_file_path[0],
+            inputNumEntries, inputSeedValue)
+
+        # Invoke the methods which will handle acquiring/downloading the data from
+        # MAST and perform the alignment
+        percentSuccess = 0.0
+        try:
+            percentSuccess = self.align_randomFields (randomCandidateTable)
+        except Exception:
+            pass
+
+        return(percentSuccess)
+
+    def align_randomFields(self, randomTable):
+        """ Process randomly selected fields (aka datasets) stored in an Astropy table.
+
+            Each field is used as input to determine if it can be aligned to an
+            astrometric standard.  The success or fail status for each test is retained
+            as the overall success or fail statistic is the necessary output from
+            this test.
+        """
+
+        numSuccess = 0
+        numQualSuccess = 0
+        numUnsuccessful = 0
+        numException = 0
+        numProcessedDatasets = 0
```

Comment:
This is defined in Line 76, so this line should not be needed.

Reply:
OK - fixed.